RESTserver/src/main/java/com/softserve/if072/restservice/service/HistoryService.java

```diff
@@ -84,12 +87,47 @@ public void deleteAll(int userId) {
         LOGGER.info(deleteAllSuccessfullyOperation, count, userId);
     }
 
+    /**
+     * Make request to a History DAO for retrieving all history records
+     * with specific product id for current user
+     *
+     * @param userID    - current user unique identifier
+     * @param productID - product unique identifier
+     * @return list of history records or empty list
+     */
     public List<History> getByProductId(int userID, int productID) {
         List<History> histories = historyDAO.getByProductId(userID, productID);
         LOGGER.info(historyContainsRecords, "product", productID, histories.size());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "product" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)